synapse/rest/media/v1/storage_provider.py

```diff
@@ -130,11 +124,11 @@ def store_file(self, path, file_info):
         if not os.path.exists(dirname):
             os.makedirs(dirname)
 
-        return defer_to_thread(
+        return await defer_to_thread(
             self.hs.get_reactor(), shutil.copyfile, primary_fname, backup_fname
         )
 
-    def fetch(self, path, file_info):
+    async def fetch(self, path, file_info):
```

Comment:
I'm confused why this needs to be an async function?

Reply:
It does not *have* to be, but the API says it is async, so it is good to be consistent.

Note that the S3 provider is truly async: https://github.com/matrix-org/synapse-s3-storage-provider/blob/b4b6525be8d5af6356c11687cde09144d2ebf8a4/s3_storage_provider.py#L100-L108